File: python/kode program/ekosistem.py
# ============================================
# Kelas Ekosistem menyimpan beberapa wilayah
# ============================================
# Kelas Ekosistem berisi kumpulan wilayah
class Ekosistem:
    def __init__(self, nama):
        self.nama = nama
        self.wilayah_list = []

    def tambah_wilayah(self, wilayah):
        self.wilayah_list.append(wilayah)

    def tampilkan(self):
        print(f"\n##### {self.nama} #####")
        for w in self.wilayah_list:
            w.tampilkan()

File: java/kode program/Ekosistem.java
import java.util.ArrayList;

// Ekosistem berisi beberapa Wilayah
public class Ekosistem {
    private String nama;
    private ArrayList<Wilayah> wilayahList = new ArrayList<>();

    public Ekosistem(String nama) {
        this.nama = nama;
    }

    public void tambahWilayah(Wilayah w) {
        wilayahList.add(w);
    }

    public void tampilkan() {
        System.out.println("=== Ekosistem: " + nama + " ===");
        for (Wilayah w : wilayahList) {
            w.tampilkan();
        }
    }
}
